src/server/array_func.c

```diff
@@ -908,6 +884,31 @@ create_subjob(job *parent, char *newjid, int *rc)
 		*rc = PBSE_IVALREQ;
 		return NULL;
 	}
+
+	/* save job and server within single transaction */
+	pbs_db_begin_trx(conn, 0, 0);
+
+	if (job_save_db(subj,SAVEJOB_NEW) != 0) {
+		job_purge(subj);
+		(void) pbs_db_end_trx(conn, PBS_DB_ROLLBACK);
+		*rc = PBSE_SAVE_ERR;
+		return ((job *)0);
+	}
+	/* save server here as part of the transaction */
+	if (svr_save_db(&server, SVR_SAVE_QUICK) != 0) {
+		job_purge(subj);
```

Comment:
Seems we no server attribute is changed here but you are calling svr_save_db(), is this really required?

Reply:
The num_jobs field inside the server table is more of a vestige. We can remove the field, or at least for now, remove the reliance on it. That way, you can remove the warning printed at server startup, and thus also not require to save the svr table each time you add a new subjob.

Reply:
We can refactor this in future to remove the num_jobs and num_queues from the svr_table, but for now we can remove the warning emitted by server when the number of jobs in the job table does not match the num_jobs field in server table. We do not need to baby sit a database :)

Reply:
@subhasisb Taking your suggestion I have removed the dependency on server.sv_qs.sv_numjobs during server recovery, and hence also removed all the svr_save_db() happening after job db commit. This change is checked in here: rc 14 https://github.com/PBSPro/pbspro/pull/590/commits/d40069815cb9a61d90e33d269a637c402573473e